pkg/dashboard/resource/function.go

```diff
@@ -103,14 +103,64 @@ func (fr *functionResource) GetByID(request *http.Request, id string) (restful.A
 	return fr.functionToAttributes(function[0]), nil
 }
 
-// Create deploys a function
+// Create and deploy a function
 func (fr *functionResource) Create(request *http.Request) (id string, attributes restful.Attributes, responseErr error) {
+	functionInfo, responseErr := fr.getFunctionInfoFromRequest(request)
+	if responseErr != nil {
+		return
+	}
+
+	// validate there are no 2 functions with the same name
+	getFunctionsOptions := &platform.GetFunctionsOptions{
+		Name:      functionInfo.Meta.Name,
+		Namespace: fr.getNamespaceFromRequest(request),
+	}
+
+	projectNameFilter := functionInfo.Meta.Labels["nuclio.io/project-name"]
```

Comment:
won't this panic if doesn't exist?? you should do `projectNameFilter, found := functionInfo.Meta.Labels["nuclio.io/project-name"]` as well as a check for empty string

Reply:
it won't panic if it doesn't exist - it'll just return an empty string

Reply:
right, so change to `projectNameFilter, found := ` :)

Reply:
okay okay